src/java.base/share/classes/sun/nio/fs/PollingWatchService.java

```diff
@@ -247,6 +249,9 @@ void update(long lastModified, int tickCount) {
      * directory and queue keys when entries are added, modified, or deleted.
      */
     private class PollingWatchKey extends AbstractWatchKey {
+
+        private static final int POLLING_INIT_DELAY = 1;
```

Comment:
Can you move this up to the top of the file so it's with DEFAULT_POLLING_INTERVAL? I thin both will need a short one line comment. 

I'm not sure about changing it to 1s as it may lead to complaints from macOS users with programs that monitor directories with a large number of files. The right thing is of course to include a native implementation for that platform, some day :-)

Reply:
> Can you move this up to the top of the file so it's with DEFAULT_POLLING_INTERVAL? I thin both will need a short one line comment.

The constants are used in different objects, so I'm not sure there is a sensible way to to do this. A constant reference to a private object might be a way, but that felt like too much indirection.

> I'm not sure about changing it to 1s as it may lead to complaints from macOS users with programs that monitor directories with a large number of files. The right thing is of course to include a native implementation for that platform, some day :-)

Changed to 2s interval between subsequent polls since there have been a few concerns about this :-). I also agree that a native implementation would be far superior.

Reply:
I think it would be cleaner to have both constants declared at the top in PollingWatchService. They are in the same nest so accessibility should not be an issue.

Reply:
My mistake. I was thinking of them as two totally separate classes. I have made this change.